pinot-query-planner/src/test/java/org/apache/pinot/query/QueryCompilationTest.java

```diff
@@ -385,10 +385,15 @@ public void testGetTableNamesForQuery() {
     assertEquals(tableNames.get(0), "a");
   }
 
-  @Test(expectedExceptions = RuntimeException.class)
+  @Test
   public void testDuplicateWithAlias() {
     String query = "WITH tmp AS (SELECT * FROM a LIMIT 1), tmp AS (SELECT * FROM a LIMIT 2) SELECT * FROM tmp";
-    _queryEnvironment.getTableNamesForQuery(query);
+    try {
+      _queryEnvironment.getTableNamesForQuery(query);
+      fail("Expected compilation of query with duplicate CTE alias to throw an exception");
+    } catch (RuntimeException e) {
+      assertTrue(e.getCause().getMessage().contains("Duplicate alias in WITH: 'tmp'"));
+    }
```

Comment:
nit: You can use `@Test(expectedExceptionsMessageRegExp = "regexp")` which will give you a better error message

Reply:
Answered [here](https://github.com/apache/pinot/pull/13664#discussion_r1684468878).